cirq/circuits/circuit_operation_test.py

```diff
@@ -307,6 +307,17 @@ def test_string_format():
 [                         ]"""
     )
 
+    fc0_global_phase = cirq.FrozenCircuit(cirq.GlobalPhaseOperation(-1))
```

Comment:
can you please add a test for the multiplication of `GlobalPhaseOperation`s? Also for nested cases.

Reply:
Expanded the test to include nested as well as unnested phase multiplication